lazyflow/operators/classifierOperators.py

```diff
@@ -417,60 +415,40 @@ def setupOutputs(self):
                                                 #ilastik operators, setting it to 2 causes errors in pixel classification
                                                 #(live prediction doesn't work when only two labels are present)
 
+        self.PMaps.meta.assignFrom( self.Image.meta )
         self.PMaps.meta.dtype = numpy.float32
-        self.PMaps.meta.axistags = copy.copy(self.Image.meta.axistags)
         self.PMaps.meta.shape = self.Image.meta.shape[:-1]+(nlabels,) # FIXME: This assumes that channel is the last axis
         self.PMaps.meta.drange = (0.0, 1.0)
 
     def execute(self, slot, subindex, roi, result):
         classifier = self.Classifier.value
         
         # Training operator may return 'None' if there was no data to train with
-        skip_prediction = (classifier is None)
+        if classifier is None:
+            result[:] = 0.0
+            return result
 
         # Shortcut: If the mask is totally zero, skip this request entirely
-        if not skip_prediction and self.PredictionMask.ready():
+        mask = None
+        if self.PredictionMask.ready():
             mask_roi = numpy.array((roi.start, roi.stop))
-            mask_roi[:,-1:] = [[0],[1]]
+            num_channels_in_mask = self.PredictionMask.meta.getTaggedShape()['c']
+            mask_roi[:,-1:] = [[0],[num_channels_in_mask]]
             start, stop = list(map(tuple, mask_roi))
-            mask = self.PredictionMask( start, stop ).wait()
-            skip_prediction = not numpy.any(mask)
-
-        if skip_prediction:
-            result[:] = 0.0
-            return result
+            multichannel_mask = self.PredictionMask( start, stop ).wait()
 
-        assert issubclass(type(classifier), LazyflowPixelwiseClassifierABC), \
-            "Classifier is of type {}, which does not satisfy the LazyflowPixelwiseClassifierABC interface."\
-            "".format( type(classifier) )
-
-        upstream_roi = (roi.start, roi.stop)
-        # Ask for the halo needed by the classifier
-        axiskeys = self.Image.meta.getAxisKeys()
-        halo_shape = classifier.get_halo_shape(axiskeys)
-        assert len(halo_shape) == len( upstream_roi[0] )
-        assert halo_shape[-1] == 0, "Didn't expect a non-zero halo for channel dimension."
-
-        # Expand block by halo, then clip to image bounds
-        upstream_roi = numpy.array( upstream_roi )
-        upstream_roi[0] -= halo_shape
-        upstream_roi[1] += halo_shape
-        upstream_roi = getIntersection( upstream_roi, roiFromShape(self.Image.meta.shape) )
-        upstream_roi = numpy.asarray( upstream_roi )
+            #create a single-channel merged mask, which has 0 iff all PredictionMask channels are 0
```

Comment:
I would have just required users to use single channel images, but this is so much better (just imaging using multiple, stacked atlasses at the same time...)